homeassistant/components/nightscout/config_flow.py

```diff
@@ -10,49 +10,56 @@
 from homeassistant.const import CONF_URL
 
 from .const import DOMAIN  # pylint:disable=unused-import
+from .utils import hash_from_url
 
 _LOGGER = logging.getLogger(__name__)
 
 DATA_SCHEMA = vol.Schema({vol.Required(CONF_URL): str})
 
 
-async def _validate_input(data):
-    """Validate the user input allows us to connect."""
-
-    try:
-        api = NightscoutAPI(data[CONF_URL])
-        status = await api.get_server_status()
-    except (ClientError, AsyncIOTimeoutError, OSError):
-        raise CannotConnect
-
-    # Return info to be stored in the config entry.
-    return {"title": status.name}
-
-
 class ConfigFlow(config_entries.ConfigFlow, domain=DOMAIN):
     """Handle a config flow for Nightscout."""
 
     VERSION = 1
     CONNECTION_CLASS = config_entries.CONN_CLASS_CLOUD_POLL
 
+    async def _validate_input(self, data):
```

Comment:
This method doesn't use `self`. Why did we move it inside the class to a method?

Reply:
Oh, my first solution did use `self`, I forgot to change it back haha